src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -661,6 +661,11 @@ static void* VirtualReserveInner(size_t size, size_t alignment, uint32_t flags,
         pRetVal = pAlignedRetVal;
     }
 
+#ifdef MADV_DONTDUMP
+    // Do not include reserved memory in coredump.
+    madvise(pRetVal, size, MADV_DONTDUMP);
```

Comment:
This should be only called when `pRetVal` is non-null.

Reply:
good catch, fixed now.